test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,860 @@
+                expect(persistenceSpy.called).to.equal(true);
+            });
+
+            it('should force a session end when override is used', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+                const mpInstance = mParticle.getInstance();
+                const persistenceSpy = sinon.spy(
+                    mpInstance._Persistence,
+                    'update'
+                );
+
+                mpInstance._SessionManager.endSession(true);
+
+                expect(mpInstance._Store.sessionId).to.equal(null);
+                expect(mpInstance._Store.dateLastEventSent).to.equal(null);
+                expect(mpInstance._Store.sessionAttributes).to.eql({});
+
+                // Persistence isn't necessary for this feature, but we should test
+                // to see that it is called in case this ever needs to be refactored
+                expect(persistenceSpy.called).to.equal(true);
+            });
+
+            it('should end log StartingEndSession message and return early if Persistence is undefined', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns(null);
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // This path currently goes through the !cookies path,
+                // which doesn't really return anything except for logging the
+                // initial StartingEndSession message
+                expect(consoleSpy.getCall(0).firstArg).to.equal(
+                    Messages.InformationMessages.StartingEndSession
+                );
+
+                // Should only log a single verbose log
+                expect(consoleSpy.getCalls().length).to.equal(1);
+            });
+
+            it('should log a NoSessionToEnd Message if Persistence Exists but does not return an sid', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and NoSessionToEnd messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.NoSessionToEnd
+                );
+            });
+
+            it('should log an AbandonedEndSession message if unable to log', () => {
```

Comment:
it's a little bit clear the use of log at the end of this test statement.  maybe the following is more explicit
```suggestion
            it('should log an AbandonedEndSession message if SDK canLog() returns false', () => {
```